### PR TITLE
exclude internal folders from lint

### PR DESCRIPTION
**Pull request by PaliC**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #106291

Title

### CODE REVIEW

File: .lintrunner.toml [hunk 1]
```diff
@@ -300,7 +309,11 @@ command = [
 [[linter]]
 code = 'NOQA'
 include_patterns = ['**/*.py', '**/*.pyi']
-exclude_patterns = ['caffe2/**']
+exclude_patterns = [
+    'caffe2/**',
+    'fb/**',
+    '**/fb/**'
```
Comment (ZainRizvi) [inline .lintrunner.toml:315]: I see various combinations of these three patterns added to all the linters.

1. Why do the exclusions added vary?  Aren't they supposed to exclude the same set of folders?
2. Any reason to not add this exclusion pattern to every linter?  That'll make them more consistent and give anyone adding a new linter in the future an easy pattern to copy/paste
3. Might there be a "universal exclusion" pattern in lintrunner already that can be leveraged?  And if not, maybe it's worth adding one to reduce the chances of human error down the road?

**Review by ZainRizvi — APPROVED** [review]
Approving, pending answers to my questions (and current changes do no harm)

File: .lintrunner.toml (hunk 1 above)
Comment (PaliC) [reply .lintrunner.toml:315]: 1) More for conciseness. If (ie. if `'fb/**'` is not needed given the inclusions, its noisy to add it imo). I just added what was needed given the inclusion rules as its a good habit to have.
2) Same as 1.
3) I agree, but that change got rejected :P https://github.com/suo/lintrunner/pull/54

**Pull request closed without merging**